src/key-value-pairs/internal.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+import styles from './styles.css.js';
+import { KeyValuePairsProps } from './interfaces';
+import { InfoLinkLabelContext } from '../internal/context/info-link-label-context';
+import ColumnLayout from '../column-layout/internal';
+import Box from '../box/internal';
+
+export { KeyValuePairsProps };
+
+const InternalKeyValuePairs = React.forwardRef(
+  ({ columns, className, ...rest }: KeyValuePairsProps, ref: React.Ref<HTMLDivElement>) => {
+    return (
+      <div {...rest} className={clsx(styles['key-value-pairs'], className)} ref={ref}>
+        {!!columns?.length && (
```

Comment:
If `columns` is required, why do you need `?.`? Why do you need this condition at all?

Reply:
updated